cmake/modules/AddSwift.cmake

```
@@ -519,12 +519,14 @@ function(_add_swift_runtime_link_flags target relpath_to_lib_dir bootstrapping)
       # Add the SDK directory for the host platform.
       target_link_directories(${target} PRIVATE "${sdk_dir}")
 
-      # A backup in case the toolchain doesn't have one of the compatibility libraries.
-      # We are using on purpose `add_dependencies` instead of `target_link_libraries`,
-      # since we want to ensure the linker is pulling the matching archives
-      # only if needed
-      target_link_directories(${target} PRIVATE "${compatibility_libs_path}")
-      add_dependencies(${target} ${compatibility_libs})
+      if(NOT "${compatibility_libs_path}" STREQUAL "")
```

Nit: there is [a more idiomatic way](https://cmake.org/cmake/help/latest/command/if.html#variable) to express this

```suggestion
      if(compatibility_libs_path)
```

---

If you have cycles, could you surrounds [the lines for the `BOOTSTRAPPING` case](https://github.com/apple/swift/pull/73619/files#diff-4b74006b7c38d950627c4183690f1c6b5435d1b4da7a327fdf6ed9bcbb2cc33bR544-R545) (lines 544-545 after the changes) with this very check?

---

Sure, fixed this, thanks!